src/orion/core/utils/__init__.py

```diff
@@ -91,7 +91,15 @@ def __init__(cls, names, bases, dictionary):
                       entry_point.dist.project_name, entry_point.dist.version)
 
         # Get types visible from base module or package, but internal
-        cls.types = cls.__base__.__subclasses__()
+        def get_all_subclasses(parent):
+            """Get subclasses recursively"""
+            subclasses = []
+            for subclass in parent.__subclasses__():
+                subclasses += [subclass] + get_all_subclasses(subclass)
```

Comment:
So I have a minor thing that there will be duplicates entries with the same class here. So we may use set here (classes are hashable), and a `list(get_all_subclasses(cls.__base__))`.